### PR TITLE
[Feature] Debit card

### PR DESCRIPTION
**Pull request by Sinu5oid**
**Context**
* Debit card module is to be introduced

**What was done**
* Added .editorconfig for the whole project
* Added EURS-WETH price helper
* Added pages (manage -- active, pending, set-email, etc)
* Added Vuex store
* Added vue-the-mask package
* Added form step guard (to be used later for refactoring)
* Added localized texts
* Updated nav items to be given `active` class instead of `exact-active` on condition
* Added new assets
* Added `SearchSkinModal` (a modal to be used the same way as `SearchTokenModal` but for skins)
* Enabled CSS source maps
* Added two new feature flags (to be able to perform rolling updates): `isDebitCardTopUpEnabled`, `isDebitCardChangeSkinEnabled`. Both are `false` atm and they should be

**How to test**
* Enable `isDebitCardEnabled` feature flag
* There should be a specific banner on the home page for the card (if only feature flag is `true`, the module is initialized and card state is `order_now`)
* Card navigation item in the home page nav menu should be clickable
* The rest of the flow is described in `TODO / Notes` section
* Optionally, one could test `Change Skin` logic and the boilerplate of `Top Up`

**TODO / Notes**
* The card module state machine manages its state by these rules:
  * If there is a hash stored, use it to identify the user and their debit card account
    * If a hash is invalid (or at least API tells us so), drop the persist item and repeat the whole process from the beginning (should not cause logical loop here)
    * Else map info response to the appropriate Vuex state to control displayable entities and flow
    
      If the state is `order_now` then proceed with the order form
      If the state is `validate_phone` then proceed with the phone validation form. The optional step is `change_phone` which triggers the service call to append a new phone number to the Trastra account. Succeeding this step will return the user to the order flow, `validate_phone` step
      Otherwise, consider the card to be active, display all the info we have at the moment. Nothing fancy here, just some card data (until the next debit card update is released)
  * Else request a hash to be created (display a form with email field)
    * Make a signature request to bind an address with an email, create persisted entry, use email for ordering if needed, otherwise never ask for email for about 1 month of the inactive period
    * Send this signature to the service API to identify the user so only they will be capable of fetching their data, nobody else won't
  * If the state can be recovered from local info, do it. Otherwise, go to the API
  * Every API request returns a valid status which could be used to modify module internal state

### CODE REVIEW

File: src/components/debit-card/manage/debit-card-manage-active.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    <div class="content">
+      <div class="container">
+        <debit-card-image :skin="currentSkin" />
+      </div>
+      <statement-list class="container">
+        <statement-list-item
+          :description="$t('debitCard.lblLast4Digits')"
+          :value="
+            cardInfo ? cardInfo.last4Digits : $t('debitCard.lblNotAvailable')
```
Comment (trukach000) [inline src/components/debit-card/manage/debit-card-manage-active.vue:13]: May be it's better to move these things to the computed property?

File: src/components/debit-card/manage/debit-card-manage-set-email.vue [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { email, required } from 'vuelidate/lib/validators';
+import { mapActions, mapGetters } from 'vuex';
+
+import { DebitCardApiError } from '@/services/mover/debit-card';
+import {
+  isProviderRpcError,
```
Comment (trukach000) [inline src/components/debit-card/manage/debit-card-manage-set-email.vue:91]: I suggest use `User-Defined Type Guards` techic to reduce the `as ProviderRpcError` lines
```export const isProviderRpcError = (
  error: unknown
): error is ProviderRpcError => {```

File: src/components/debit-card/manage/debit-card-manage-validate-phone.vue [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      if (this.$v.$invalid) {
+        this.scrollButtonIntoView();
+        return;
+      }
+
+      try {
+        this.isLoading = true;
+        await this.validatePhoneNumber(this.code);
+        this.isLoading = false;
+      } catch (error) {
```
Comment (trukach000) [inline src/components/debit-card/manage/debit-card-manage-validate-phone.vue:192]: Probably we can move this.IsLoading to the `finally` section

File: src/components/debit-card/manage/debit-card-manage-active-skeleton.vue [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  <secondary-page :title="$t('debitCard.lblBeautifulCard')">
+    <p class="description">{{ $t('debitCard.txtVisaDebitCard') }}</p>
+
+    <div class="content">
+      <div class="container">
+        <pu-skeleton class="card-image skeleton" tag="div" />
+      </div>
+      <statement-list class="container">
+        <li v-for="idx in 4" :key="idx" class="item">
+          <pu-skeleton tag="h3" width="190px" />
```
[the remark this answers is not in the document]
Comment (Sinu5oid) [reply src/components/debit-card/manage/debit-card-manage-active-skeleton.vue:11]: As statement-list is to be refactored in near future along other components, I haven't made any skeleton alternative for this use case
This is the main reason I used plain `<pu-skeleton>` component and its props to control the appearence

To be refactored after feature release, I guess

File: src/views/debit-card/debit-card-change-skin.vue [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    },
+    selectorStyle(): CssProperties {
+      if (this.skinToBeApplied === undefined) {
+        return {
+          backgroundColor: '#f1f1f1',
+          boxShadow: '0 0 8px rgb(0, 0, 0, 0.5)'
+        };
+      }
+
+      const assetColor = this.skinToBeApplied.color;
```
[the remark this answers is not in the document]
Comment (Sinu5oid) [reply src/views/debit-card/debit-card-change-skin.vue:142]: It is used twice, once as object field directly,  second time along with string interpolation. I don't really see any benefit achieved by that

File: src/components/debit-card/manage/debit-card-manage-change-phone.vue [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            ) as string;
+            this.scrollButtonIntoView();
+            this.isLoading = false;
+            return;
+          }
+        }
+
+        this.errorText = this.$t('debitCard.errors.default') as string;
+        this.scrollButtonIntoView();
+        this.isLoading = false;
```
[the remark this answers is not in the document]
Comment (Sinu5oid) [reply src/components/debit-card/manage/debit-card-manage-change-phone.vue]: I believe that keeping logical blocks as close as possible is the key for understanding large code controlled by branching
Keeping
```ts
handle() {
  this.errorMessage = 'something';
  this.scrollButtonIntoView();
  this.isLoading = false;
}
```
works better than
```ts
handle() {
  this.errorMessage = 'something'; // as a default
  /*
  * 50-ish lines of alternative code
  */
  // continue with default
  this.scrollButtonIntoView();
  this.isLoading = false;
}
```

**Review by McMannaman — APPROVED** [review]
LGTM 🙏
Let's perform final tests

**Pull request merged**